google/gax/__init__.py

```diff
@@ -482,3 +491,157 @@ def __next__(self):
+            of.
+        any_pb (google.protobuf.any_pb2.Any): the object to be converted.
+
+    Returns:
+        An instance of the pb_type message.
+    """
+    msg = pb_type()
+    if any_pb.Unpack(msg):
+        return msg
+
+    raise TypeError(
+        'Could not convert {} to {}'.format(
+            any_pb.__class__.__name__, pb_type.__name__))
+
+
+class _OperationFuture(object):
+    """A Future which polls a service for completion via OperationsClient."""
+
+    def __init__(self, operation, client, result_type, metadata_type,
+                 call_options=None):
+        """Constructor.
+
+        Args:
+            operation (google.longrunning.Operation): the initial long-running
+                operation object.
+            client (google.gapic.longrunning.operations_client.OperationsClient):
+                a client for the long-running operation service.
+            result_type (type): the class type of the result.
+            metadata_type (type): the class type of the metadata.
+            call_options (google.gax.CallOptions, optional): the call options
+                that are used when reloading the operation.
+        """
+        self._last_operation = operation
+        self._client = client
+        self._result_type = result_type
+        self._metadata_type = metadata_type
+        self._call_options = call_options
+        self._done_clbks = []
+        self._process = None
+
+    def cancel(self):
+        """If last Operation's value of `done` is true, returns false;
+           otherwise, issues OperationsClient.cancel_operation and returns true.
+        """
+        if self.done():
+            return False
+
+        self._client.cancel_operation(self._last_operation.name)
+        return True
+
+    def result(self, timeout=None):
+        """Enters polling loop on OperationsClient.get_operation, and once
+           Operation.done is true, then returns Operation.response if successful
+           or throws GaxError if not successful.
+
+           This method will wait up to timeout seconds. If the call hasn't
+           completed in timeout seconds, then a concurrent.futures.TimeoutError
+           will be raised. timeout can be an int or float. If timeout is not
+           specified or None, there is no limit to the wait time.
+        """
+        if not self._poll(timeout).HasField('response'):
+            raise GaxError(self._last_operation.error.message)
+
+        return _from_any(self._result_type, self._last_operation.response)
+
+    def exception(self, timeout=None):
+        """Similar to result(), except returns the exception if any."""
+        if self._poll(timeout).HasField('error'):
+            return self._last_operation.error
+
+        return None
+
+    def cancelled(self):
+        """Return True if the call was successfully cancelled."""
+        self._get_operation()
+        return self._last_operation.HasField('error') and \
```

Comment:
nit: using a line continuation is a little weird, instead of something like the following:

```python
return (self._last_operation.HasField('error') and
    self._last_operation.error.code == code_pb2.CANCELLED)
```